L1Trigger/L1TGlobal/plugins/L1TGlobalProducer.cc

```diff
@@ -50,54 +31,65 @@
 #include "CondFormats/L1TObjects/interface/L1GtParameters.h"
 #include "CondFormats/DataRecord/interface/L1GtParametersRcd.h"
 
-/* //These are old but we might be able to reuse when these go into for upgrade
-#include "CondFormats/L1TObjects/interface/L1GtPrescaleFactors.h"
-#include "CondFormats/DataRecord/interface/L1GtPrescaleFactorsAlgoTrigRcd.h"
-#include "CondFormats/L1TObjects/interface/L1GtTriggerMask.h"
-#include "CondFormats/DataRecord/interface/L1GtTriggerMaskAlgoTrigRcd.h"
-#include "CondFormats/DataRecord/interface/L1GtTriggerMaskVetoAlgoTrigRcd.h"
-*/
-
 #include "DataFormats/Common/interface/RefProd.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/MessageLogger/interface/MessageDrop.h"
 
+#include "L1Trigger/L1TGlobal/plugins/L1TGlobalProducer.h"
 
+using namespace l1t;
 
 // constructors
 
-l1t::GtProducer::GtProducer(const edm::ParameterSet& parSet) :
-            m_muInputTag(parSet.getParameter<edm::InputTag> ("GmtInputTag")),
-            m_caloInputTag(parSet.getParameter<edm::InputTag> ("caloInputTag")),
-	    m_extInputTag(parSet.getParameter<edm::InputTag> ("extInputTag")),
+void L1TGlobalProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  // These parameters are part of the L1T/HLT interface, best not to change:
+  desc.add<edm::InputTag> ("GmtInputTag");
+  desc.add<edm::InputTag> ("CaloInputTag");
+  desc.add<edm::InputTag> ("ExtInputTag");
+  desc.add<bool>("AlgorithmTriggersUnprescaled");
+  desc.add<bool>("AlgorithmTriggersUnmasked");
+  // These parameters are defaulted, and are not part of the L1T/HLT interface
```

Comment:
Yes! Please!
(Even if those values are non-working dummy placeholders [as they must be specified in any used python instance, one would specify proper values there]. But the fillDescriptions also needs values as otherwise ConfDB template parsing will not work)